### PR TITLE
[DISCO-2531] GitHub landing page

### PR DESCRIPTION
**Pull request by taddes**
## References

JIRA: [DISCO-2531](https://mozilla-hub.atlassian.net/browse/DISCO-2531)
GitHub: [#162](https://github.com/mozilla-services/consvc-shepherd/issues/162)

## Description

The Service Overview page in GitHub should contain the following information:

    Architecture Diagram. Ideally the diagram is in Mermaid for ease of change and maintainability.

    High level description of what the purpose of the service is.

    Links to API documentation

    What datastores should we be aware of? This can also just be part of the architecture diagram.

    What external dependencies should we be aware of? (i.e. Shepherd etc.). This can also just be part of the architecture diagram.

## PR Review Checklist

_Put an `x` in the boxes that apply_

- [x] This PR conforms to the [Contribution Guidelines](https://github.com/mozilla-services/consvc-shepherd/blob/main/CONTRIBUTING.md).
- [x] The PR title starts with the JIRA issue reference, format `[DISCO-####]`, and has the same title (if applicable).
- [ ] [Documentation](https://github.com/mozilla-services/consvc-shepherd/tree/main/docs) has been updated (if applicable).
- [ ] Functional and performance test coverage has been expanded and maintained (if applicable).

[DISCO-2531]: https://mozilla-hub.atlassian.net/browse/DISCO-2531?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mtbottle — CHANGES_REQUESTED** [review]
The Quick Start Guide will need to be reworked a bit. In particular, there are too many "options" that it's providing users for their environment set up (which makes it not really "quick"). I'm not sure if you would prefer to have that as a separate ticket, or also just make the changes in this PR. I think [this comment](https://github.com/mozilla-services/contile/pull/614#discussion_r1268278020) is particularly relevant on stuff to fix for the guide.

Also, just a hint (in case you don't know about it), but you can use https://mermaid.live/edit to work on your architecture diagram with up-to-date preview! I find that it's easier to work this way, then copy it back into the documentation. I would be interested to hear if you're using another mermaid live updating platform!

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# Contextual Services Shepherd
+# Contextual Services Shepherd [![CircleCI](https://dl.circleci.com/status-badge/img/gh/mozilla-services/consvc-shepherd/tree/main.svg?style=shield)](https://dl.circleci.com/status-badge/redirect/gh/mozilla-services/consvc-shepherd/tree/main)
 
-This is a tool to manage dynamic settings for Contextual Services projects like
+Shepherd is a tool to manage dynamic settings for Contextual Services projects like
 Merino and Contile. The settings managed here can be changed at runtime,
```
[the remark this answers is not in the document]
Comment (taddes) [reply README.md:4]: This was in this README when I moved it, so I know there is a hope for this one day, but agreed, it is a bit misleading at this point.

File: docs/quickStart.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+Ex:
+```shell
+export PATH="$HOME/.pyenv/bin:$PATH"
+eval "$(pyenv init -)"
+eval "$(pyenv virtualenv-init -)"
+```
+
+You also need to ensure you have Postgres installed. You can select a distribution from
+[https://www.postgresql.org/download/](https://www.postgresql.org/download/), use homebrew
+or their EDB installer client.  Make sure to install version 14.7, or equivalent, and have
```
[the remark this answers is not in the document]
Comment (taddes) [reply docs/quickStart.md:21]: I'll make a note to talk with Tif about this, as to my knowledge, we both use this DB setup instead of Docker. We may need to make some revisions to the Docker setup given the progress on Shepherd. Why don't we leave this for now and I've made a note to revisit.  While setup with Postgres does work quite well, but we definitely want a smooth Docker option for sure. 👍

File: docs/intro.md [hunk 3]
```diff
@@ -1,107 +1,37 @@
...
-Merino and Contile. The settings managed here can be changed at runtime,
-reducing the need for deployments of the services that use it.
+## Service Overview
 
-## Quick Start
+Shepherd is a Django application that allows users to make changes to Contile's advertising settings without redeploying Contile.  The settings managed here can be changed at runtime, eliminating the need for deployments of the services that use it.
 
-To use consvc-shepherd, you'll need a Python 3.11 development environment and
-Poetry installed. To achieve this, it is recommended you use a Python virtual environment.
+Within Shepherd, users can make changes to Advertising Partners. A given Partner can have multiple advertisers in numerous geolocations, each with their own distinct domain.  Users can also define partner allocation percentages for each tile position. This means that given each Contile tile position, you can set the percentage of impressions that will show from each partner.
```
[the remark this answers is not in the document]
Comment (taddes) [reply docs/intro.md:7]: This is great, thanks for sharing! Really insightful article on it.

File: docs/quickStart.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (tiftran) [reply docs/quickStart.md]: Sooo I don't actually use the DB setup anymore, currently running shepherd locally and db within docker. My perspective is that I don't think it really matters how we currently have shepherd set up. I think providing the easiest starting point for someone to set up should be sufficient and it would be up to them at that point to decide whatever variations they prefer. @taddes @mtbottle thoughts?

[the remark this answers is not in the document]
Comment (mtbottle) [reply docs/quickStart.md]: > I think providing the easiest starting point for someone to set up should be sufficient and it would be up to them at that point to decide whatever variations they prefer.

I would agree with that. I think we should go further and discourage variations. The reasons for this are:

1. Less experienced users may not have enough knowledge to know which set up is easier, especially if we present them with many alternative options.
2. It's easier to find help if everyone had the same set up.
3. Alternative set-ups make it difficult to know if errors are due to bad set up, or something actually wrong with the service.
4. If the service set up is standardized, then people will be more proactive about maintaining the set up documentation, rather than deferring it believing that it's only an isolated "local set up problem". Problems are more likely to be raised and fixed.

That said, I think moving the database to Docker is the right move. Docker makes it easier to:

1. destroy + recreate a database on whim (i.e can be more experimental with testing migrations and other stuff)
2. isolate your Postgres database (i.e. will there be potentially a naming conflict with database, schema or user?)
3. easily allow for varying different Postgres versions on the same machine (i.e. services depending on different versions of Postgres can have the version that makes the most sense for that service). 

tl;dr is that we should stick with one set up, and that we should use Docker for bootstrapping any service resources if possible.

[the remark this answers is not in the document]
Comment (tiftran) [reply docs/quickStart.md]: Oh, I meant I agree that there should be one set up method in our docs. I double checked, the docker compose file still works for me. 

As a aside, I think engineers being able to customize their set up is an interesting topic that could be discussed in our next 1:1? (or as a team), but I don't think it blocks this PR.

[the remark this answers is not in the document]
Comment (mtbottle) [reply docs/quickStart.md]: >  I think engineers being able to customize their set up is an interesting topic that could be discussed in our next 1:1? (or as a team?)

Interesting! I've added it as an agenda item for our tech sync up. Good idea thanks! 😄

[the remark this answers is not in the document]
Comment (taddes) [reply docs/quickStart.md]: Agreed! Sounds great to me @mtbottle & @tiftran. Given our meeting discussion, I'll make the changes so we opt for the Docker setup and will remove the other alternate/superfluous steps here. I'll just check to make sure there aren't any caveats with the Docker setup we need to annotate.

File: docs/quickStart.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+DB_NAME=postgres
+DB_USER=postgres
+DB_HOST=db
+DB_PASS=postgres
+```
+
+5. Set up and enable your virtual environment:
+
+```shell
+# pyenv verison install
```
Comment (mtbottle) [inline docs/quickStart.md:38]:
```suggestion
# pyenv version install
```

File: docs/UserGuide.md [hunk 5]
```diff
@@ -43,5 +44,6 @@ Click the Add buttons for Settings snapshots under CONSVC_SHEPHERD:
 5. Select “Publish Settings Snapshot” from the actions dropdown and click “Go”.
    * A green status bar should appear at the top of the page stating the snapshot has been published.
 
-*Note that Shepherd is unable to undo snapshots, if changes need to be undone, one would have to manually revert the advertiser information and create a new snapshot.
+*Note that Shepherd is unable to undo snapshots. 
+If changes need to be undone, one would have to manually revert the advertiser information and create a new snapshot.
```
Comment (mtbottle) [inline docs/UserGuide.md:48]: No actions necessary but just a note: We can probably like the runbook here once it's written.

File: docs/intro.md [hunk 6]
```diff
@@ -1,107 +1,46 @@
...
-use `.env.example` and rename it to `env`:
-
-```shell
-$ mv .env.example .env
+## Architecture
+```mermaid
+flowchart LR
+    AdopsUser[\fa:fa-user AdOps User/] --> |Update allocation &<br/>settings snapshots| Shep(Shepherd) --> DB[(Postgres DB)]
+    Shep --> |Validated JSON snapshot<br/>sent to GC bucket for access.| GCP[Google Cloud JSON]
+    GCP -..- |Periodic sync for<br/>tile information| Contile{Contile}
```
Comment (mtbottle) [inline docs/intro.md:23]: Let's use the same shape for the services (Shepherd and Contile).
```suggestion
    GCP -..- |Periodic sync for<br/>tile information| Contile(Contile)
```

**Review by mtbottle — APPROVED** [review]
Just a few minor comment, but feel free to merge once they're addressed! Thanks for making the instructions clearer 🚀 :shipit:

File: docs/intro.md (hunk 6 above)
Comment (taddes) [reply docs/intro.md:23]: Thanks @mtbottle ! 🚀
